src/main/java/com/commercetools/project/sync/state/StateSyncer.java

```diff
@@ -77,7 +75,16 @@ public static StateSyncer of(
   @Nonnull
   @Override
   protected CompletionStage<List<StateDraft>> transform(@Nonnull List<State> states) {
-    return toStateDrafts(getSourceClient(), referenceIdToKeyCache, states);
+    return toStateDrafts(getSourceClient(), referenceIdToKeyCache, states)
+        .handle(
+            (drafts, exception) -> {
+              if (exception != null) {
+                final SyncException syncException = new SyncException(exception.getCause());
+                logErrorCallback(LOGGER, "state", syncException, Optional.empty(), List.of());
```

Comment:
This change is a workaround because in java-sync-sdk v2 the SDK StateDraftBuilder.build() throws error when key is missing. This might be removed / adjusted when https://commercetools.atlassian.net/browse/DEVX-277 is fixed. :)

I have made it consistent here: [751492a](https://github.com/commercetools/commercetools-project-sync/pull/512/commits/751492a07f7620bacd765df1d8f7236bf8b84a8b)